frontend/views/containers/contributions/IncomeDetails.vue

```diff
@@ -0,0 +1,262 @@
+            :userpledgeamount='canPledge && !!form.pledge ? Number(form.pledge) : null'
+            :userincomeamount='needsIncome && !!form.income ? Number(form.income) : null'
+          )
+</template>
+
+<script>
+import { mapGetters } from 'vuex'
+import { validationMixin } from 'vuelidate'
+import { required } from 'vuelidate/lib/validators'
+import sbp from '~/shared/sbp.js'
+import InputAmount from './InputAmount.vue'
+import PaymentMethods from './PaymentMethods.vue'
+import Tooltip from '@components/Tooltip.vue'
+import ModalBaseTemplate from '@components/Modal/ModalBaseTemplate.vue'
+import TextWho from '@components/TextWho.vue'
+import GroupPledgesGraph from '../GroupPledgesGraph.vue'
+import currencies from '@view-utils/currencies.js' // TODO make filter
+import L from '@view-utils/translations.js'
+
+export default {
+  name: 'IncomeDetails',
+  mixins: [validationMixin],
+  components: {
+    ModalBaseTemplate,
+    InputAmount,
+    TextWho,
+    Tooltip,
+    PaymentMethods,
+    GroupPledgesGraph
+  },
+  data () {
+    return {
+      ephemeral: {
+        formFeedbackMsg: {}
+      },
+      form: {
+        incomeDetailsKey: null,
+        amount: null
+      },
+      fakeStore: {
+        incomeDetailsKey: 'pledgeAmount', // incomeAmount || pledgeAmount
+        amount: 100
+      }
+    }
+  },
+  computed: {
+    ...mapGetters([
+      'groupSettings',
+      'memberProfile'
+    ]),
+    memberGroupProfile () {
+      const profile = this.memberProfile(this.$store.state.loggedIn.username) || {}
+      return profile.groupProfile || {}
+    },
+    currency () {
+      return currencies[this.groupSettings.mincomeCurrency]
+    },
+    infoRequired () {
+      return this.L('This information is required.')
+    },
+    needsIncome () {
+      return this.form.incomeDetailsKey === 'incomeAmount'
+    },
+    canPledge () {
+      return this.form.incomeDetailsKey === 'pledgeAmount'
+    },
+    inputAmountError () {
+      if (this.needsIncome && !this.$v.form.amount.hasLowIncome) {
+        return this.L('If your income is the same or higher than the group\'s mincome change your answer to "Yes, I do"')
+      } else if (this.$v.form.amount.$error) {
+        return this.infoRequired
+      }
+      return ''
+    }
+  },
+  created () {
+    const incomeDetailsKey = this.memberGroupProfile.incomeDetailsKey
+    if (incomeDetailsKey) {
+      this.form.incomeDetailsKey = incomeDetailsKey
+      this.form.amount = this.memberGroupProfile[incomeDetailsKey]
+    }
+  },
+  methods: {
+    verifyInputIncome (e) {
+      this.form.income = e.target.value
+      this.$v.form.income.$touch()
+    },
+    verifyInputPledge (e) {
+      this.form.pledge = e.target.value
+      this.$v.form.pledge.$touch()
+    },
+    resetAmount () {
+      this.form.amount = this.$v.form.incomeDetailsKey.$model === this.memberGroupProfile.incomeDetailsKey ? this.memberGroupProfile[this.memberGroupProfile.incomeDetailsKey] : ''
```

Comment:
`.$model` is only for binding to `v-model` (it causes `.touch()` to be called automatically). Outside of that it's simpler to directly access the variable, in this case, `this.form.incomeDetailsKey`

Reply:
once again, you are reviewing the old version :(